gateway/src/apicast/policy/url_rewriting/policy.lua

```diff
@@ -0,0 +1,74 @@
+--- URL rewriting policy
+-- This policy allows to modify the path of a request.
+
+local ipairs = ipairs
+local sub = ngx.re.sub
+local gsub = ngx.re.gsub
+
+local policy = require('apicast.policy')
+local _M = policy.new('URL rewriting policy')
+
+local new = _M.new
+
+local substitute_functions = { sub = sub, gsub = gsub }
+
+-- func needs to be ngx.re.sub or ngx.re.gsub.
+-- This method simply calls one of those 2. They have the same interface.
+local function substitute(func, subject, regex, replace, options)
+  local new_uri, num_changes, err = func(subject, regex, replace, options)
+
+  if not new_uri then
+    ngx.log(ngx.WARN, 'There was an error applying the regex: ', err)
+  end
+
+  return new_uri, num_changes > 0
+end
+
+-- Returns true when the URL was rewritten and false otherwise
+local function apply_rewrite_command(command)
+  local func = substitute_functions[command.op]
+
+  if not func then
+    ngx.log(ngx.WARN, "Unknown URL rewrite operation: ", command.op)
+  end
+
+  local new_uri, changed = substitute(
+    func, ngx.var.uri, command.regex, command.replace, command.options)
+
+  if changed then
+    ngx.req.set_uri(new_uri)
+  end
+
+  return changed
+end
+
+--- Initialize a URL rewriting policy
+-- @tparam[opt] table config. Contains the rewrite commands.
```

Comment:
the dot after config looks like extra. You can see it at https://3244-53965666-gh.circle-artifacts.com/0/doc/modules/policy.url_rewriting.policy.html

Reply:
Well spotted. Fixed.